### PR TITLE
DAOS-10530 agent: Fix NUMA node rotation (#8973)

### PR DESCRIPTION
**Pull request by kjacque**
The agent was assuming NUMA nodes were sorted by index rather than
ID for load balancing purposes. This broke in cases with network
interfaces on non-sequential NUMA nodes.

Features: control

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

**Comment by kjacque** [comment]
This one was not a clean cherry-pick.

**Comment by kjacque** [comment]
The single test failure is a known intermittent failure, [DAOS-7193](https://daosio.atlassian.net/browse/DAOS-7193). Given that this is an intermittent issue, I will restart from the hardware test stage.

File: src/control/cmd/daos_agent/fabric.go [hunk 1]
```diff
@@ -247,20 +248,31 @@ func (n *NUMAFabric) getNextDevice(numaNode int) *FabricInterface {
...
-		numa := (n.defaultNumaNode + i) % numNodes
-		fi, err := n.getDeviceFromNUMA(numa, netDevClass, provider)
+		n.currentNUMANode = (n.currentNUMANode + 1) % numNodes
+		fi, err := n.getDeviceFromNUMA(nodes[n.currentNUMANode], netDevClass, provider)
 		if err == nil {
 			// Start the search on this node next time
-			n.defaultNumaNode = numa
-			n.log.Debugf("Suitable fabric interface %q found on NUMA node %d", fi.Name, numa)
+			n.log.Debugf("Suitable fabric interface %q found on NUMA node %d", fi.Name,
+				nodes[n.currentNUMANode])
```
Comment (knard-intel) [inline src/control/cmd/daos_agent/fabric.go:260]: Strange that there is not the same value from the original patch. From what I understand, the debug message from the patch is probably invalid: `n.currentNUMANode` replace in this patch with `nodes[n.currentNUMANode]`.

**Review by knard-intel — COMMENTED** [review]
OK for me.
Just a minor difference with the original patch which could have to be fixed in the original one.

File: src/control/cmd/daos_agent/fabric.go (hunk 1 above)
Comment (kjacque) [reply src/control/cmd/daos_agent/fabric.go:260]: I think I forgot to change it in the original patch. I had to go in manually on this one to resolve conflicts and ended up noticing this.

**Comment by kjacque** [comment]
Huh. In fact this was a different issue when I dug into it. See: https://daosio.atlassian.net/browse/DAOS-10050

The failing test is normally skipped in 2.0 due to this failure, I think it got picked up due to the fact that I was running with parameters in Jenkins to boost the priority. I included the control tag, which pulled this test in.

As this test is known to fail on 2.0, can we force-land?

**Pull request merged**